src/astra_driver.cpp

```diff
@@ -181,13 +199,16 @@ void AstraDriver::getAvailabilityDiagnostic(diagnostic_updater::DiagnosticStatus
 
   if (device_count == 0)
   {
-    stat.summary(DiagnosticStatus::ERROR, "No Device connected");
+    diagnostics_updater_.broadcast(DiagnosticStatus::ERROR, "No Device connected");
     ROS_WARN("NO DEVICE CONNECTED, RESTARTING");
 
     device_.reset();
 
     terminateROSTopics();
     initDevice(); // blocking
+
+    imageConnectCb();
```

Comment:
Can you comment on what motivated this?

Reply:
Our current implementation works with always active depth and color streams, which means we want to publish everything as long as device is connected. Here, after the device is lost, re-plugged and initialized again, no depth/color stream will be started unless e.g. we have a new subscriber. By executing a connect callback again here we ensure that the streams are correctly started **and** the frequency diagnostics are started to be `ticked` at every frame so that the proper frequency will be shown.